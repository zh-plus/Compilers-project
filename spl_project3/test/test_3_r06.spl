int hanoi(int n, int p1, int p2, int p3){
    if(n == 1){
        write(p1*1000000+p3);
    }
    else{
        hanoi(n-1,p1,p3,p2);
        write(p1*1000000+p3);
        hanoi(n-1,p2,p1,p3);
    }
    return 0;
}
int main(){
    int sum = 3;
    hanoi(sum, 1, 2, 3);
    return 0;
}

//1000003
//1000002
//3000002
//1000003
//2000001
//2000003
//1000003