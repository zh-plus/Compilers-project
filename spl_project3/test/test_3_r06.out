FUNCTION hanoi :
PARAM t0
PARAM t1
PARAM t2
PARAM t3
IF t0 == #1 GOTO label1
GOTO label2
LABEL label1 :
t4 := t1 * #1000000
t5 := t4 + t3
WRITE t5
GOTO label3
LABEL label2 :
t7 := t0 - #1
ARG t2
ARG t3
ARG t1
ARG t7
t8 := CALL hanoi
t9 := t1 * #1000000
t10 := t9 + t3
WRITE t10
t12 := t0 - #1
ARG t3
ARG t1
ARG t2
ARG t12
t13 := CALL hanoi
LABEL label3 :
RETURN #0
FUNCTION main :
t14 := #3
ARG #3
ARG #2
ARG #1
ARG t14
t15 := CALL hanoi
RETURN #0
